eureka-client/src/main/java/com/netflix/discovery/endpoint/EndpointUtils.java

```diff
@@ -145,10 +145,18 @@ public void randomize(List<String> urlList) {
                     randomizer.randomize(ec2Urls);
                 }
                 for (String ec2Url : ec2Urls) {
-                    String serviceUrl = "http://" + ec2Url + ":"
-                            + clientConfig.getEurekaServerPort()
-                            + "/" + clientConfig.getEurekaServerURLContext()
-                            + "/";
+                    StringBuilder sb = new StringBuilder()
+                            .append("http://")
+                            .append(ec2Url)
+                            .append(":")
+                            .append(clientConfig.getEurekaServerPort());
+                    if (clientConfig.getEurekaServerURLContext() != null) {
+                        if (!clientConfig.getEurekaServerURLContext().startsWith("/")) {
+                            sb.append("/");
+                        }
+                        sb.append(clientConfig.getEurekaServerURLContext());
+                    }
+                    String serviceUrl = sb.toString();
```

Comment:
Hi. To replicate the existing behaviour, when the context is not null, this should append an additional `/`.

Reply:
You are right. Added commit to append slash.